source/extensions/filters/http/credential_injector/credential_injector_filter.cc

```diff
@@ -0,0 +1,150 @@
+#include "source/extensions/filters/http/credential_injector/credential_injector_filter.h"
+
+#include "source/common/http/headers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CredentialInjector {
+
+FilterConfig::FilterConfig(CredentialInjectorSharedPtr credential_injector, bool overwrite,
+                           bool allow_request_without_credential, const std::string& stats_prefix,
+                           Stats::Scope& scope)
+    : injector_(credential_injector), overwrite_(overwrite),
+      allow_request_without_credential_(allow_request_without_credential),
+      stats_(generateStats(stats_prefix + "credential_injector.", scope)) {}
+
+// Inject configured credential to the HTTP request header.
+// return true if successful
+bool FilterConfig::injectCredential(Http::RequestHeaderMap& headers) {
+  absl::Status status = injector_->inject(headers, overwrite_);
+
+  // credential already exists in the header and overwrite is false
```

Comment:
nit: please use full sentences, e.g. `If the credential already exists..., increment the counter.`